src/charm.py

```diff
@@ -69,9 +78,27 @@ def __init__(self, *args):
         self.store_requirer = ParcaStoreEndpointRequirer(
             self, relation_name="external-parca-store-endpoint"
         )
+        # Enable charm tracing
+        self.charm_tracing = TracingEndpointRequirer(
+            self, relation_name="charm-tracing", protocols=["otlp_http"]
+        )
+        self.charm_tracing_endpoint, self.server_cert = charm_tracing_config(
+            self.charm_tracing, CA_CERT_PATH
+        )
+
+        self.framework.observe(self.on.parca_pebble_ready, self._configure_and_start)
```

Comment:
is this part of another refactoring task? if so, we're missing a piece which is to map events to individual handlers.
```suggestion
        self.framework.observe(self.on.parca_pebble_ready, self._on_parca_pebble_ready)
```

Reply:
and, `_configure_and_start` feels like a candidate to be turned into a holistic `_reconcile()` method.

Reply:
reverted. Will do the refactoring in a separate PR.